### PR TITLE
Remove misleading example from no-return-await

### PR DESCRIPTION
**Pull request by jmrossy**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
The example in the Correct Code section that marks a function as `async` without using `await` is a bit misleading.

It incorrectly implies that a function must itself be async if it's simply returning a promise without ever awaiting it. In fact, doing so would slightly hurt performance without any benefit.

**Is there anything you'd like reviewers to focus on?**
Nope, small change :)

### CODE REVIEW

**Comment by mdjermanovic** [comment]
Thanks for the PR!

I agree that it might be better to remove this example.

**Comment by platinumazure** [comment]
Thanks for the PR!

I think the intent of that example is to show that a return statement without an await will not violate this rule. In that sense, I think it's a good example.

Could we instead add a comment in the example, to address the issue that you're concerned about?

**Comment by platinumazure** [comment]
Hi @jmrossy, go ahead and update at your convenience :smile:

@mdjermanovic Do you have any concerns with my suggestion to just add a clarifying comment to the documentation example?

**Comment by jmrossy** [comment]
@platinumazure Done :)

**Comment by mdjermanovic** [comment]
> @mdjermanovic Do you have any concerns with my suggestion to just add a clarifying comment to the documentation example?

Actually, after reading #6820 my opinion is now quite different, and I completely agree that the example should stay. My first thoughts were that an example that would be reported by a best practice category rule should not be present in another rule's docs. The rule in question is `require-await`. While the rule can be used to find possible errors, I believe it's a valid use to mark a function with `async` even if it doesn't change anything, just to make it clear that the function always returns a Promise.

File: docs/rules/no-return-await.md [hunk 1]
```diff
@@ -36,6 +32,15 @@ async function foo() {
...
 }
+
+async function foo() {
+    /** 
+     * Note: Though this is not a lint rule violation, 
+     * this function simply proxies a value and does not 
+     * need to be async  
+     */
+    return bar();
+}
```
Comment (mdjermanovic) [inline docs/rules/no-return-await.md]: This isn't necessarily true, it depends on `bar`. `foo` always returns a Promise and also never throws, it doesn't have to be the case with `bar`.

This code could be used to ensure that the call does not throw, or to wrap the returned non-promise value from 'bar'.

Comment (jmrossy) [reply docs/rules/no-return-await.md]: Ok. I've removed the comment. Seems like we've come full circle here so I guess I'll settle for just having the atypical example moved to the bottom.

**Comment by kaicataldo** [comment]
Do you mind updating the commit message and PR title to more accurately reflect the change?

File: docs/rules/no-return-await.md [hunk 2]
```diff
@@ -36,6 +32,10 @@ async function foo() {
         return await bar();
     } catch (error) {}
 }
+
+async function foo() {
+    return bar();
+}
 ```
 
 In the last example the `await` is necessary to be able to catch errors thrown from `bar()`.
```
Comment (mdjermanovic) [inline docs/rules/no-return-await.md]: A problem with the change is that this sentence now does not apply to the 'last' example. I'm not sure what would be the best way to fix this.

Comment (jmrossy) [reply docs/rules/no-return-await.md]: Ah yeah, good catch. Okay, the value of this change is officially not worth the trouble it has created for you guys. Closing it, thanks for your time :)

**Pull request closed without merging**